lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -1592,31 +1592,68 @@ namespace Js
 #endif
 #endif
 
+
 #if DYNAMIC_INTERPRETER_THUNK
 #ifdef _M_IX86
+    __declspec(naked)
+        Var InterpreterStackFrame::AsmJsDelayDynamicInterpreterThunk(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        __asm
+        {
+            push ebp;
+            mov ebp, esp;
+            push[esp + 8];    // push function object
+            call WasmLibrary::EnsureWasmEntrypoint;
+            test eax, eax;
+            jne skipThunk;
+
+            push[esp + 8];    // push function object
+            call InterpreterStackFrame::EnsureDynamicInterpreterThunk;
+skipThunk:
+#ifdef _CONTROL_FLOW_GUARD
+            // verify that the call target is valid
+            push eax;
+            mov  ecx, eax;
+            call[__guard_check_icall_fptr];
+            pop eax;
```

Comment:
CFG check restores ecx. rather than saving/restoring from the stack it's better to do mov ecx, eax

Reply:
done and I've looked for other places where we do this as well

Reply:
thanks for cleaning that up, it looks good to me!